web/src/app/components/bar-chart/bar-chart.component.ts

```diff
@@ -318,21 +333,33 @@ export class BarChartComponent implements AfterViewInit {
       if (!this.focus) { return; }
       this.focus.selectAll('rect').remove();
 
-      const newBars = this.focus.selectAll('rect')
-        .data(inputData);
-      // TODO: Call function that creates a boxplot.
+      const newBars = this.focus.selectAll('rect').data(inputData);
       const maxFocusDuration = d3.max(inputData, (d: Item) => d.durationHours);
       if (!maxFocusDuration) { return; }
 
       // Upate the xScaleFocus, yScaleFocus, xAxisFocus and yAxisFocus
       // based on the selected data for the focus chart.
       this.xScaleFocus.domain(inputData.map((d: Item) => d.startTime));
       this.yScaleFocus.domain([0, maxFocusDuration]);
+
       if (!this.yAxis) { return; }
-      this.yAxis.transition().duration(0).call(d3.axisLeft(this.yScaleFocus));
+      this.yAxis.call(d3.axisLeft(this.yScaleFocus));
+
+      // If the focus chart has more than 100 bars, set the ticks and labels
+      // on xAxisFocus to be always smaller than 100.
       if (!this.xAxisFocus) { return; }
-      this.xAxisFocus.transition().duration(0)
-        .call(d3.axisBottom(this.xScaleFocus).tickSizeOuter(0))
+      if (inputData.length > BarChartComponent.DEFAULT_MAX_LABELS) {
+        const modNum = Math.round((inputData.length / BarChartComponent.DEFAULT_MAX_LABELS));
+        this.xAxisFocus
+          .call(d3.axisBottom(this.xScaleFocus)
+            .tickValues(this.xScaleFocus.domain().filter((d: Item, i: number) => !(i % modNum)))
+            .tickSizeOuter(0));
```

Comment:
```suggestion
            .tickSizeOuter(0)
          );
```
align with opening parenthesis